### PR TITLE
[CPF-3838] Add TTLCachingDAO support to EasyDAO

### PR DESCRIPTION
**Pull request by KolombetM**
https://nanopay.atlassian.net/browse/CPF-3838

### CODE REVIEW

File: src/services [hunk 1]
```diff
@@ -328,7 +328,7 @@ p({
...
-p({"class":"foam.nanos.boot.NSpec", "name":"menuDAO",                            "serve":true,  "serviceScript":"return new foam.dao.EasyDAO.Builder(x).setName(\"menu\").setPm(true).setJournalType(foam.dao.JournalType.SINGLE_JOURNAL).setJournalName(\"menus\").setOf(foam.nanos.menu.Menu.getOwnClassInfo()).setOrder(new foam.core.PropertyInfo[] {foam.nanos.menu.Menu.ORDER, foam.nanos.menu.Menu.LABEL}).build().orderBy(foam.nanos.menu.Menu.ORDER);", "client":"{\"of\":\"foam.nanos.menu.Menu\",\"cache\":true}"})
+p({"class":"foam.nanos.boot.NSpec", "name":"menuDAO",                            "serve":true,  "serviceScript":"return new foam.dao.EasyDAO.Builder(x).setName(\"menu\").setPm(true).setJournalType(foam.dao.JournalType.SINGLE_JOURNAL).setJournalName(\"menus\").setOf(foam.nanos.menu.Menu.getOwnClassInfo()).setOrder(new foam.core.PropertyInfo[] {foam.nanos.menu.Menu.ORDER, foam.nanos.menu.Menu.LABEL}).build().orderBy(foam.nanos.menu.Menu.ORDER);", "client":"{\"of\":\"foam.nanos.menu.Menu\",\"cacheType\":\"FULL\"}"})
```
Comment (jlhughes) [inline src/services]: Pretty print please.

**Comment by jlhughes** [comment]
And from slack: 
Update TTLCachingDAO and make mergeDelay a property and also have mergeDelay be a property in EasyDAO 
caching: enum
cacheMergeDelay: long

File: src/foam/dao/EasyDAO.js [hunk 2]
```diff
@@ -694,6 +696,11 @@ foam.CLASS({
             of: this.model
           });
         }
+        if ( this.cacheType == foam.dao.CacheType.TTL ) {
```
Comment (kgrgreer) [inline src/foam/dao/EasyDAO.js]: You could make this an 'else if'.

File: src/foam/dao/TTLCachingDAO.js [hunk 3]
```diff
@@ -23,7 +23,11 @@ foam.CLASS({
     {
       /** The cache to read items quickly. */
       name: 'cache'
-    }
+    },
+    {
+      class: 'Long',
+      name: 'purgeTime'
+    },
```
Comment (jlhughes) [inline src/foam/dao/TTLCachingDAO.js:31]:
```suggestion
      value: 15000
    },
```

File: src/services [hunk 4]
```diff
@@ -328,7 +328,24 @@ p({
...
 p({"class":"foam.nanos.boot.NSpec", "name":"historyDAO",                         "serve":true,  "serviceScript":"return new foam.dao.PMDAO(x, new foam.dao.java.JDAO(x, foam.dao.history.HistoryRecord.getOwnClassInfo(), \"historyRecords\"));","client":"{\"of\":\"foam.dao.history.HistoryRecord\"}"})
-p({"class":"foam.nanos.boot.NSpec", "name":"menuDAO",                            "serve":true,  "serviceScript":"return new foam.dao.EasyDAO.Builder(x).setName(\"menu\").setPm(true).setJournalType(foam.dao.JournalType.SINGLE_JOURNAL).setJournalName(\"menus\").setOf(foam.nanos.menu.Menu.getOwnClassInfo()).setOrder(new foam.core.PropertyInfo[] {foam.nanos.menu.Menu.ORDER, foam.nanos.menu.Menu.LABEL}).build().orderBy(foam.nanos.menu.Menu.ORDER);", "client":"{\"of\":\"foam.nanos.menu.Menu\",\"cache\":true}"})
+p({
+  "class": "foam.nanos.boot.NSpec",
+  "name" :"menuDAO",
+  "serve" :true,
+  "serviceScript":
+  """
+    return new foam.dao.EasyDAO.Builder(x)
+      .setName(\"menu\")
```
Comment (jlhughes) [inline src/services]:
```suggestion
      .setName("menu")
```

File: src/services [hunk 5]
```diff
@@ -328,7 +328,24 @@ p({
...
+      .setPm(true)
+      .setJournalType(foam.dao.JournalType.SINGLE_JOURNAL)
+      .setJournalName(\"menus\")
```
Comment (jlhughes) [inline src/services]:
```suggestion
      .setJournalName("menus")
```

File: src/services [hunk 6]
```diff
@@ -328,7 +328,24 @@ p({
...
+      .setOf(foam.nanos.menu.Menu.getOwnClassInfo())
+      .setOrder(new foam.core.PropertyInfo[] {foam.nanos.menu.Menu.ORDER, foam.nanos.menu.Menu.LABEL})
+      .build()
+      .orderBy(foam.nanos.menu.Menu.ORDER);
```
Comment (jlhughes) [inline src/services]: remove, duplicate.

Comment (KolombetM) [reply src/services]: Do you want me to remove 4 lines and then duplicate them?

File: src/foam/dao/TTLCachingDAO.js (hunk 3 above)
Comment (KolombetM) [reply src/foam/dao/TTLCachingDAO.js:31]: Should default value for purgeTime be removed from EasyDAO then?

File: src/services (hunk 6 above)
Comment (jlhughes) [reply src/services]: just the orderBy line. - just remove that.

Comment (jlhughes) [reply src/services]: it duplicates the setOrder line.

**Review by jlhughes — APPROVED** [review]

**Pull request merged**